crates/biome_js_analyze/tests/specs/nursery/useJsxCurlyBraceConvention/invalid.jsx

```diff
@@ -0,0 +1,5 @@
+<Foo>{'Hello world'}</Foo>;
```

Comment:
I let you decide how to do this: we should add more tests that involve comments. E.g.:

```jsx
<Foo>{ /*comment*/ 'Hello world'/*comment*/ }</Foo>
```

Reply:
I'd rather go ahead and fix it in this PR. I'm a little bit stumped on how to actually apply the trivia.

This gives me the trivia as `SyntaxTrivia<JsLanguage>`
```rust
let l_brace_trivia = expr.l_curly_token().map(|c| c.trailing_trivia()).ok()?;
```
But `JsSyntaxToken::new_detached` expects an iterator of `TriviaPiece`.

Reply:
We have many rules that work on trivia, you can take some inspiration from them. Generally, working with trivia is a pain, just so you know! 🤣 

In this case, go straight with this:

```rs
let trivia = expr.l_curly_token().ok()?.trailing_trivia().pieces();
```

`.pieces()` gives you an iterator over trivia. When you have `pieces`, there are few methods that work with them e.g. `prepend_trivia_pieces`, `append_trivia_pieces`.

There's also a method called `chain_trivia_pieces` that merges two iterators.

Reply:
You're right, working with trivia is a pain. I've attempted to use `add_jsx_elements_before/after_element` to preserve comments, like in the snippet you posted, and it doesn't seem to be working. As far as I can tell, I'm using the functions correctly, but those changes are not showing up in the diff.

Reply:
My personal advice is the **closely** work [with the playground](https://biomejs.dev/playground/?code=PABGAG8AbwA%2BAHsAIAAvACoAYwBvAG0AbQBlAG4AdAAqAC8AIAAnAEgAZQBsAGwAbwAgAHcAbwByAGwAZAAnAC8AKgBjAG8AbQBtAGUAbgB0ACoALwAgAH0APAAvAEYAbwBvAD4A). Notice that the comment is attached to the **trailing** slot?

![Screenshot 2024-07-04 at 16 27 22](https://github.com/biomejs/biome/assets/602478/148c1e33-0912-4cd1-ae87-b0a7d829942b)

There are a pair of `[] []`, the first one is the slot for leading trivia, the second slot is for trailing trivia.

Reply:
Ah, I see what you're doing. That won't work. The way rowan works, pointers **can't exist** if there isn't a node that already exists. In your particular case, you are attaching trivia to a token that doesn't exist already. Hence, you must create a **detached node**. Below an example of how a new token is created from scractch, and we attach trivia to it:

https://github.com/biomejs/biome/blob/6584de11976f6edf626146b20df9074816b3bcbc/crates/biome_analyze/src/lib.rs#L669-L674

Reply:
Should be fixed now. I can add more tests if needed.